server/src/main/java/org/elasticsearch/common/util/set/Sets.java

```diff
@@ -165,11 +175,12 @@ public static <T> Set<T> intersection(Set<T> set1, Set<T> set2) {
      *
      * @param set      set to copy
      * @param elements elements to add
+     * @return the unmodifiable copy of the input set with the extra elements added
      */
     @SuppressWarnings("unchecked")
     public static <E> Set<E> addToCopy(Set<E> set, E... elements) {
         final var res = new HashSet<>(set);
         Collections.addAll(res, elements);
-        return Set.copyOf(res);
+        return (Set<E>) Set.of(res.toArray());
```

Comment:
This will create 1 additional copy - can you please revert to the original code?

Reply:
`Set.copyOf(res)` = `Set.of(new HashSet<>(res).toArray());`

So it's the original code that was creating an additional copy here.

Reply:
My bad, I thought Set.copyOf was more intelligent and I did not check. Thanks!